### PR TITLE
sys/Makefile.include: include riotboot headers when FEATURES_REQUIRED=riotboot

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

Currently when wanting to use riotboot on any application `riotboot_slot` module has to be included manually either by passing as make argument or editing the Makefile itself. Because of this if following [riotboot/README](https://github.com/RIOT-OS/RIOT/blob/209d90bc00458501ae710148d3cf0e1b59faff8b/bootloaders/riotboot/README.md) you will end up with `/bin/sh: 1: arithmetic expression: expecting primary: " + "` error. 

Since whenever someone uses `riotboot` feature it needs `riotboot_slot` modules (either for the bootloader or the application) it could be included as a dependency. 

### Testing procedure

Run:

`BOARD=samr21-xpro FEATURES_REQUIRED+=riotboot APP_VER=$(date +%s) make -C examples/hello-world riotboot/flash-combined-slot0`

Works with this PR, On master it fails with error:

```
"make" -C /home/francisco/workspace/RIOT/sys/newlib_syscalls_default
"make" -C /home/francisco/workspace/RIOT/sys/pm_layered
"make" -C /home/francisco/workspace/RIOT/sys/stdio_uart
/bin/sh: 1: arithmetic expression: expecting primary: " + "
```

### CODE REVIEW

**Comment by fjmolinas** [comment]
@cladmi Is it incorrect to use `FEATURES_REQUIRED` as a filter to include a MODULE, regarding all the changes being tackled in #9913.

**Comment by kaspar030** [comment]
> Since whenever someone uses `riotboot` feature it needs `riotboot_slot` modules (either for the bootloader or the application) it could be included as a dependency.

The cause are probably missing slot defines.
Can check see if this patch also solves the issue?

https://gist.github.com/ddedc16fa9927df6727bc97d8b3ed2c6

**Comment by fjmolinas** [comment]
> Can  check see you check if this patch also solves the issue?

@kaspar030 Nope, it doesn't fix the issue.

**Comment by kaspar030** [comment]
> @kaspar030 Nope, it doesn't fix the issue.

weird, it does for me. just double checked. I changed that line in current master, are we on the same branch?

**Comment by fjmolinas** [comment]
@kaspar030 You are right found the mistake. Should have directly applied the patch, I hadn't removed the trailing underscore.

**Comment by fjmolinas** [comment]
@kaspar030 thanks for the pointers! Changed the PR to use your patch!

**Comment by fjmolinas** [comment]
@kaspar030 All checks are green!

File: sys/Makefile.include [hunk 1]
```diff
@@ -77,7 +77,7 @@ ifneq (,$(filter printf_float,$(USEMODULE)))
   endif
 endif
 
-ifneq (,$(filter riotboot_%,$(USEMODULE)))
+ifneq (,$(filter riotboot%,$(FEATURES_USED)))
```
Comment (kaspar030) [inline sys/Makefile.include]: I think the ```%``` at the end of riotboot is not necessary. If you feel like, could you confirm and then force-push its removal? I'm OK with merging as is, too. ;)

Comment (fjmolinas) [reply sys/Makefile.include]: yep!

**Comment by kaspar030** [comment]
&go. Thanks!

**Pull request merged**